packages/@react-stately/color/src/useHexColorFieldState.ts

```diff
@@ -0,0 +1,142 @@
+  const [isValid, setIsValid] = useState(isInputValueValid(colorValue, maxValue, minValue));
+
+  const increment = () => {
+    setColorInputValue((previousValue) => {
+      const color = typeof previousValue === 'string'? new Color(previousValue) : previousValue;
+      const colorString = color.toString('hex').substring(1);
+      const colorNumber = parseInt(colorString, 16);
+
+      const maxColor = typeof maxValue === 'string'? new Color(maxValue) : maxValue;
+      const maxColorString = maxColor.toString('hex').substring(1);
+      const maxColorNumber = parseInt(maxColorString, 16);
+
+      const newValue = `#${Math.min(colorNumber + step, maxColorNumber).toString(16).toUpperCase()}`;
+      const newColor = new Color(newValue);
+      updateValidation(newColor);
+      setInputValue(newValue);
+      return newColor;
+    });
+  };
+
+  const incrementToMax = useCallback(() => {
+    if (maxValue != null) {
+      const maxColor = typeof maxValue === 'string'? new Color(maxValue) : maxValue;
+      setColorInputValue(maxColor);
+      setInputValue(maxColor.toString('hex'));
+    }
+  }, [maxValue, setColorInputValue, setInputValue]);
+
+  const decrement = () => {
+    setColorInputValue((previousValue) => {
+      const color = typeof previousValue === 'string'? new Color(previousValue) : previousValue;
+      const colorString = color.toString('hex').substring(1);
+      const colorNumber = parseInt(colorString, 16);
+
+      const minColor = typeof minValue === 'string'? new Color(minValue) : minValue;
+      const minColorString = minColor.toString('hex').substring(1);
+      const minColorNumber = parseInt(minColorString, 16);
+
+      const newValue = `#${Math.min(colorNumber + step, minColorNumber).toString(16).toUpperCase()}`;
+      const newColor = new Color(newValue);
+      updateValidation(newColor);
+      setInputValue(newValue);
+      return newColor;
+    });
+  };
+
+  const decrementToMin = useCallback(() => {
+    if (minValue != null) {
+      const minColor = typeof minValue === 'string'? new Color(minValue) : minValue;
+      setColorInputValue(minColor);
+      setInputValue(minColor.toString('hex'));
+    }
+  }, [minValue, setColorInputValue, setInputValue]);
+
+  const setColorValue = (color: Color) => {
+    updateValidation(color);
+    setColorInputValue(color);
+    setInputValue(color.toString('hex'));
+  };
+
+  const updateValidation = (value: Color) => {
+    setIsValid(isInputValueValid(value, maxValue, minValue));
+  };
+
+  const commitInputValue = () => {
+    if (!inputValue.length) return;
+    updateValidation(colorValue);
+    setInputValue(colorValue.toString('hex'));
+  };
+
+  return {
+    colorValue,
+    setColorValue,
+    value: null,
+    setValue: null,
+    inputValue,
+    increment,
+    incrementToMax,
+    decrement,
+    decrementToMin,
+    commitInputValue,
+    validationState: !isValid ? 'invalid' : null,
```

Comment:
Need to check this with the team. I copied this from `NumberField` but this kind of suggests the component will not be in a `valid` state (with the green check icon)